packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -184,13 +151,6 @@ export class GarbageCollector implements IGarbageCollector {
 	private readonly gcOptions: IGCRuntimeOptions;
```

Comment:
I think it'd be best to remove this and the metatadata as private class fields, and just access their info through the config interface which takes these as inputs.

The only downside I see is to logging, where these are spread into `details` in a few places.  Maybe config can include a property `inputs: TelemetryProperties` or whatever the type is, that can be used for logging (and it's set to `{ ...metadata, ...gcOptions }`)

Reply:
Move `runFullGC` directly into configs. The rest of the options are used to generate configs so didn't add them. As for the logging, maybe we can log it once during load and not every time.

Reply:
Take a look now.

Reply:
Yeah I agree about only logging the full GC Options once, and then everywhere else just log config.